docs/source/tutorials/arch_modeling/user_defined_temp_tutorial.rst

```diff
@@ -0,0 +1,37 @@
+================================================================
+Integrating Custom Verilog Modules with user_defined_templates.v
+================================================================
+1. Introduction
+===============
+**In this tutorial, we will**
+- Provide motivation for generating the user_defined_templates.v verilog file
+- Go through a generated user_defined_templates.v file to demonstrate how to use it
+For this examaple, we are using a modified version of the hard adder task that comes with OpenFPGA
+To follow along, remove the path for the verilog_netlist in the file k6_frac_N10_adder_chain_40nm_openfpga.xml at line 187.
+
+.. figure:: ./figures/modified_arch_file.png
+   :scale: 50%
+   
+From the OpenFPGA root directory run the command
+``python3 openfpga_flow/scripts_run_fpga_task.py fpga_verilog/adder/hard_adder --debug --show_thread_logs``
+Running this command should fail and produce output similar to this
+
+.. figure:: ./figures/Error_log.png
+   :scale: 50%
+   
+This command failed during the verification step because the path to the module definition for ADDF is missing. In our architecture file, user-defined verilog modules are those ``<circuit_model>`` with the key term verilog_netlist. The user_defined_templates.v file provides a module template for incorporating Hard IPs with no external library into the architecture. 
```

Comment:
For the ``user_define_templates.v``, please add a link to the FPGA-Verilog documentation.